File: SocketServer.Shared/Messaging/MessageEnvelope.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.IO;

namespace SocketServer.Shared.Messaging
{
    public abstract class MessageEnvelope
    {
        protected MessageEnvelope()
        {
            MessageEncoding = Encoding.ASCII;
        }

        public Encoding MessageEncoding { get; set; }

        public abstract void Serialize(object message, Stream stream);
        public abstract object Deserialize(Stream stream);
    }
}

File: Java/BlazeServer/src/com/BlazeServer/Messaging/MessageEnvelope.java
package com.BlazeServer.Messaging;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;
import java.nio.charset.Charset;

public abstract class MessageEnvelope {
	
	private Charset messageEncoding;
	
	protected MessageEnvelope() {
		messageEncoding = Charset.defaultCharset();
	}
	
	public Charset getMessageEncoding() {
		return messageEncoding;
	}
	
	public void setMessageEncoding(Charset value) {
		messageEncoding = value;
	}
	
	public abstract void Serialize(Object message, OutputStream stream) throws IOException;
	public abstract Object Deserialize(InputStream stream) throws IOException;
}
